python/ampli/v1/ampli/ampli.py

```diff
@@ -0,0 +1,889 @@
+                  group_name: Union[str, List[str]],
+                  event_options: Optional[EventOptions] = None):
+        if not self.initialized_and_enabled():
+            return
+        if not event_options:
+            event_options = EventOptions()
+        if user_id:
+            event_options["user_id"] = user_id
+        self._client.set_group(group_type, group_name, event_options)
+
+    def revenue(self, user_id: str,
+                revenue_obj: Revenue,
+                event_options: Optional[EventOptions] = None):
+        if not self.initialized_and_enabled():
+            return
+        if not event_options:
+            event_options = EventOptions()
+        if user_id:
+            event_options["user_id"] = user_id
+        self._client.revenue(revenue_obj, event_options)
+        
+    def add(self, amp_plugin: Union[EventPlugin, DestinationPlugin]):
+        if not self.initialized_and_enabled():
+            return
+        self._client.add(amp_plugin)
+        
+    def remove(self, amp_plugin: Union[EventPlugin, DestinationPlugin]):
+        if not self.initialized_and_enabled():
+            return
+        self._client.remove(amp_plugin)
+        
+    def flush(self):
+        if not self.initialized_and_enabled():
+            return
+        self._client.flush()
+        
+    def shutdown(self):
+        if not self.initialized_and_enabled():
+            return
+        self._client.shutdown()
+        self.disabled = True
+        
+    def event_max_int_for_test(
+        self,
+        user_id: str,
+        int_max_10: int,
+        event_options: Optional[EventOptions] = None
+    ):
+        """Track event 'EventMaxIntForTest'
+
+
+        [View in Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Track%20event%20'EventMaxIntForTest')
+
+
+        Event to test schema validation
+
+        Owner: a94a8fe5ccb19ba61c4c0873d391e987982fbbd3 06b69c130b0036944180e653185c74f600c6fe43
+
+        :param user_id: The user's ID.
+        :param int_max_10: property to test schema validation
+        """
+        event = EventMaxIntForTest(
+            int_max_10
+        )
+        event['user_id'] = user_id
+        self.track(event, event_options)
+
+    def event_no_properties(
+        self,
+        user_id: str,
+        event_options: Optional[EventOptions] = None
+    ):
+        """Track event 'Event No Properties'
+
+
+        [View in Tracking Plan](https://data.amplitude.com/test-codegen/Test%20Codegen/events/main/latest/Track%20event%20'Event%20No%20Properties')
+
+
+        Event w no properties description
+
+        Owner: a94a8fe5ccb19ba61c4c0873d391e987982fbbd3 06b69c130b0036944180e653185c74f600c6fe43
+
+        :param user_id: The user's ID.
+        """
+        event = EventNoProperties(
+
+        )
+        event['user_id'] = user_id
+        self.track(event, event_options)
+
+    def event_object_types(
+        self,
+        user_id: str,
```

Comment:
do we still need user_id if we already have event_options?

Reply:
That is how we handle it in Node. Since user_id is mostly always required it was made its own param.